### PR TITLE
Fix NL/BE/DE address parsing for uppercase suffixes and hyphens

### PR DESCRIPTION
**Pull request by SeriouslyAwesome**
**Fixes:** shop/issues-checkout#13795
**Slack:** https://shopify.slack.com/archives/C01M8T03N9W/p1785249440092579

## Problem

NL/BE/DE address parsing fails on common real-world patterns, causing empty house number fields in checkout:

- ❌ **Uppercase suffixes**: `Voltastraat 2 A`, `Loswal 12 B` 
- ❌ **Hyphenated ranges**: `Philippusweg 3-5`, `Avenue Louise 12-14`
- ❌ **Streets starting with digits** (NL): `1e Helmersstraat 5`

## Root Cause

`address1_regex` patterns were too restrictive:
- streetNumber pattern `\d+(?: ?[a-z])?` only allowed **lowercase** letters
- No hyphen support
- NL streetName pattern `[^\d]+` excluded digits entirely

On regex failure, the entire address dumps into `streetName`, leaving `streetNumber` empty.

## Changes

| Pattern Component | Before | After |
|------------------|--------|-------|
| **streetNumber** | `\d+(?: ?[a-z])?` | `\d+(?:-\d+[A-Z]*)?(?: ?[A-Za-z])?` |
| **NL streetName** | `[^\d]+` | `(?:\d+[a-z]+\s+)?[^\d]+` |

### What This Allows

✅ Uppercase unit letters: `2 A`, `12 B`, `6 C`
✅ Hyphenated ranges: `3-5`, `12-14`, `12-II` (with uppercase Roman numerals)
✅ NL streets starting with ordinals: `1e Helmersstraat`, `2e Jan Steenstraat`

### Limitations

- Multi-letter suffixes like "bis", "hs" still not supported (edge case, separate issue if needed)
- Keeps existing safety: ambiguous addresses like `Kempenaar 25 11` still fallback to full string

## Testing

- ✅ All existing tests pass (87 tests)
- ✅ New tests for uppercase suffixes, hyphens, and digit-prefixed streets
- ✅ Manual verification with reported failing addresses from shop 97632256326

## Business Impact

**Shop:** 97632256326 (bouwmaat.myshopify.com, NL, Plus tier)
**Reported:** 25% of pickup locations affected, 25% drop in orders (unverified, pending data access)

## Deployment

This changes data files (`data/regions/*.yml`), so:
1. New worldwide gem version required
2. Atlas + shop/world must bump the gem
3. Checkout-web will pick up the new patterns automatically

### CODE REVIEW

File: lang/typescript/src/extended-address/splitAddress1.test.ts [hunk 1]
```diff
@@ -196,6 +224,11 @@ describe('splitAddress1', () => {
       address: '84A Rue du merlo',
       expected: {streetName: 'Rue du merlo', streetNumber: '84A'},
     },
+    // Hyphenated house numbers
```
Comment (ananyaneogi) [inline lang/typescript/src/extended-address/splitAddress1.test.ts:231]: Can we also add a test case for "12-II" (roman numerals) from your PR description

**Review by ananyaneogi — APPROVED** [review]

File: lang/typescript/src/extended-address/splitAddress1.test.ts (hunk 1 above)
Comment (SeriouslyAwesome) [reply lang/typescript/src/extended-address/splitAddress1.test.ts:231]: Good call, added!
